### ISSUE TITLE
() so it seems shadows will equip gray item they pick up to replace the level 1 heirloom items they already have

### ISSUE THREAD

**Issue by YggdrasilWotLK** [YggdrasilWotLK/Yggdrasil-WoW#141]
so it seems shadows will equip gray item they pick up to replace the level 1 heirloom items they already have

**Comment by Ye-Old-Potato** [comment]
Yeah they seem to pick up whatever seems to have the highest raw numbers without any regard for many other stats, for instance they will almost always choose the LK25hc hammer instead of Valnyr if they're healing you.

This is per my observation, I can try recording some more of this replicated if you see the need for it.

**Comment by YggdrasilWotLK** [comment]
I see your comment doesn't explicitly address heirlooms; long story short they assess the value of gear by multiplying stats in the item template and ilvl. Hierlooms have no stats in the item template and ilvl 1, so they always get a very very very low score and they will always unequip them in favor of literally anything else. The issue is here https://github.com/YggdrasilWotLK/mod-playerbots/blob/edf4b62a0aa6e832ad0bf1f53ffe580b2286f6f5/src/factory/StatsWeightCalculator.cpp#L383

**Comment by Ye-Old-Potato** [comment]
Ahhhh, that makes sense. Yeah I've never observed explicit events with heirlooms but I have seen such issues with other gear rarities.

**Comment by YggdrasilWotLK** [comment]
Closing this as we've since disabled auto gear swapping in favor of manually swapping with commans inv -> e
